projects/wave-core-new/src/lib/time/time-slider/time-slider.component.ts

```diff
@@ -0,0 +1,282 @@
+import {Component, OnInit, ChangeDetectionStrategy, ViewChild, ChangeDetectorRef, OnDestroy} from '@angular/core';
+import {DataSet} from 'vis-data/standalone';
+import {DateType, Timeline} from 'vis-timeline/standalone';
+import {ElementRef} from '@angular/core';
+import {LayoutService} from '../../layout.service';
+import {ProjectService} from '../../project/project.service';
+import moment, {DurationInputArg2, Moment} from 'moment';
+import {Subscription} from 'rxjs';
+import {Time} from '../time.model';
+import {Layer} from '../../layers/layer.model';
+
+@Component({
+    selector: 'wave-time-slider',
+    templateUrl: './time-slider.component.html',
+    styleUrls: ['./time-slider.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class TimeSliderComponent implements OnInit, OnDestroy {
+    //Timeline Data for vis-timeline
+    timeline: Timeline | undefined;
+    options: any;
+    data: any;
+    groups: any;
+
+    //Time Data
+    startTime: DateType = new Date();
+    endTime: DateType = new Date();
+    timeScale: {value: DurationInputArg2; text: string}[] = [
+        {value: 'hour', text: 'Hours'},
+        {value: 'day', text: 'Days'},
+        {value: 'month', text: 'Months'},
+        {value: 'year', text: 'Years'},
+    ];
+    selectedScale: DurationInputArg2 = 'year';
+    isRange = true;
+
+    //Layer Data
+    layerList: Array<Layer> = [];
+
+    @ViewChild('timeline', {static: true}) timelineContainer!: ElementRef;
+
+    // inventory of used subscriptions
+    private subscriptions: Array<Subscription> = [];
+
+    constructor(
+        protected readonly projectService: ProjectService,
+        protected readonly layoutService: LayoutService,
+        private changeDetectorRef: ChangeDetectorRef,
+    ) {
+        this.subscriptions.push(
+            this.projectService.getLayerStream().subscribe((layerList) => {
+                if (layerList !== this.layerList) {
+                    this.layerList = layerList;
+                }
+                this.changeDetectorRef.markForCheck();
+            }),
+        );
+    }
+
+    ngOnInit(): void {
+        this.getTimelineData();
+        this.getTimelineGroups();
```

Comment:
Isn't this a potential race condition since you expect the subscription in the constructor to be finished before you call these methods and loop over the layer list?

Reply:
Maybe that is an solution 159738b46d800332d57c4bd300aef3b9443bd526